internal/pkg/presentation/api/api.go

```diff
@@ -4,84 +4,46 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io"
 	"net/http"
-	"net/http/pprof"
 
 	"github.com/diwise/iot-agent/internal/pkg/application"
-	"github.com/diwise/iot-agent/internal/pkg/application/iotagent"
+	"github.com/diwise/iot-agent/internal/pkg/application/facades"
+	"github.com/diwise/iot-agent/internal/pkg/application/types"
+	"github.com/diwise/iot-agent/internal/pkg/presentation/api/auth"
 	"github.com/diwise/iot-agent/pkg/lwm2m"
 	"github.com/diwise/senml"
 	"github.com/diwise/service-chassis/pkg/infrastructure/o11y"
 	"github.com/diwise/service-chassis/pkg/infrastructure/o11y/logging"
 	"github.com/diwise/service-chassis/pkg/infrastructure/o11y/tracing"
-	"github.com/go-chi/chi/v5"
-	"github.com/riandyrn/otelchi"
-	"github.com/rs/cors"
+
 	"go.opentelemetry.io/otel"
 )
 
 var tracer = otel.Tracer("iot-agent/api")
 
-type API interface {
-	Router() chi.Router
-	health(w http.ResponseWriter, r *http.Request)
-}
-
-type api struct {
-	r                  chi.Router
-	app                iotagent.App
-	forwardingEndpoint string
-}
-
-type PayloadStorer interface {
-	Save(ctx context.Context, se application.SensorEvent) error
-}
+func RegisterHandlers(ctx context.Context, rootMux *http.ServeMux, app application.App, facade facades.EventFunc, policies io.Reader) error {
+	const apiPrefix string = "/api/v0"
 
-func New(ctx context.Context, r chi.Router, facade, forwardingEndpoint string, app iotagent.App, storer PayloadStorer) (API, error) {
-	return newAPI(ctx, r, facade, forwardingEndpoint, app, storer)
-}
-
-func newAPI(ctx context.Context, r chi.Router, facade, forwardingEndpoint string, app iotagent.App, storer PayloadStorer) (*api, error) {
-	a := &api{
-		r:                  r,
-		app:                app,
-		forwardingEndpoint: forwardingEndpoint,
+	authenticator, err := auth.NewAuthenticator(ctx, policies)
+	if err != nil {
+		return fmt.Errorf("failed to create api authenticator: %w", err)
 	}
 
-	r.Use(cors.New(cors.Options{
-		AllowedOrigins:   []string{"*"},
-		AllowCredentials: true,
-		Debug:            false,
-	}).Handler)
-
-	serviceName := "iot-agent"
-
-	r.Use(otelchi.Middleware(serviceName, otelchi.WithChiRoutes(r)))
-
-	r.Get("/health", a.health)
+	mux := http.NewServeMux()
 
-	r.Post("/api/v0/messages", a.incomingMessageHandler(ctx, facade, storer))
-	r.Post("/api/v0/messages/lwm2m", a.incomingLWM2MMessageHandler(ctx))
-	r.Post("/api/v0/messages/schneider", a.incomingSchneiderMessageHandler(ctx))
+	mux.HandleFunc("POST /messages", NewIncomingMessageHandler(ctx, app, facade))
```

Comment:
Check out the new router wrapper in the service chassis. It will make the chi divorce a little less painful.